src/document/mod.rs

```diff
@@ -84,113 +116,225 @@ impl DocumentEncryptOpts {
-    /// default to only sharing with the creator of the document
+    /// Constructs a `DocumentEncryptOpts` with common values.
+    ///
+    /// The document will have a generated ID and no name. Access to the document will be granted only to its creator.
     fn default() -> Self {
         DocumentEncryptOpts::with_explicit_grants(None, None, true, vec![])
     }
 }
 
 #[async_trait]
 pub trait DocumentOps {
-    /// List all of the documents that the current user is able to decrypt.
+    /// Encrypts the provided document bytes.
+    ///
+    /// This will create an entry for the document in the IronCore service. Returns a `DocumentEncryptResult`
+    /// which contains document metadata as well as the `encrypted_data`, which is the only thing that must be passed
+    /// to [document_decrypt()](trait.DocumentOps.html#tymethod.document_decrypt) in order to decrypt the document.
+    ///
+    /// # Arguments
+    /// - `document_data` - Bytes of the document to encrypt
+    /// - `encrypt_opts` - Document encryption parameters. Default values are provided with `DocumentEncryptOpts::default()`.
     ///
-    /// # Returns
-    /// `DocumentListResult` struct with vec of metadata about each document the user can decrypt.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// use ironoxide::document::DocumentEncryptOpts;
+    /// let data = "secret data".as_bytes();
+    /// let encrypted = sdk.document_encrypt(data, &DocumentEncryptOpts::default()).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_encrypt(
+        &self,
+        document_data: &[u8],
+        encrypt_opts: &DocumentEncryptOpts,
+    ) -> Result<DocumentEncryptResult>;
+
+    /// Decrypts an IronCore encrypted document.
+    ///
+    /// Returns details about the document as well as its decrypted bytes.
+    ///
+    /// # Arguments
+    /// - `encrypted_document` - Bytes of encrypted document. These should be the same bytes returned from
+    /// [document_encrypt()](trait.DocumentOps.html#tymethod.document_encrypt).
+    ///
+    /// # Errors
+    /// Fails if passed malformed data or if the calling user does not have sufficient access to the document.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let encrypted_data: Vec<u8> = vec![];
+    /// let decrypted_document = sdk.document_decrypt(&encrypted_data).await?;
+    /// let decrypted_data = decrypted_document.decrypted_data();
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_decrypt(&self, encrypted_document: &[u8]) -> Result<DocumentDecryptResult>;
+
+    /// Lists all of the IronCore encrypted documents that the calling user has been
+    /// granted access to.
```

Comment:
Maybe we just remove the "IronCore" part here. I know that your goal here was that the unmanaged documents won't be returned here, but I'm not sure if that's worth specifically calling out? I dunno, I could be swayed either way, I was just trying to make it less vague as to what an "IronCore document" meant.
```suggestion
    /// Lists all of the encrypted documents that the calling user can decrypt
```

Reply:
Yeah I want a way to make that distinction, but I'm fine with taking it out for clarity.

This is one place where "that the calling user can decrypt" technically isn't correct though (which is why I wrote a test for `document_list` in the PR). If you're the author of a document, even if you make `grant_to_author` false, it will still show up in your list.

Reply:
Ah right, that weird corner case. I'd probably be ok with ignoring that here. Alternatively something like "Lists all of the encrypted documents that the caller user can decrypt or authored" I suppose. We can wait for someone else to weigh in too.

Reply:
I think I'll try "Lists all of the encrypted documents that the calling user can read or decrypt."